### PR TITLE
Update android gradle plugins versions

### PR DESCRIPTION
**Pull request by m4gr3d**
Bump the Android gradle plugins versions to the latest.

Android Studio 4.1+ seems to stop providing editor support for (and recognizing) Android libraries when they define an external native build whose root includes the library's directory.
As a workaround, I moved the external native build definition into a separate empty Android library. This does not affect the project structure since the external native build definition was added in the first place for Android Studio editor support (the actual native dependencies are handled by `scons`).

### CODE REVIEW

File: platform/android/java/app/config.gradle [hunk 1]
```diff
@@ -6,7 +6,8 @@ ext.versions = [
     buildTools         : '30.0.1',
     supportCoreUtils   : '1.0.0',
     kotlinVersion      : '1.4.10',
-    v4Support          : '1.0.0'
+    v4Support          : '1.0.0',
+    javaVersion        : 1.8
```
Comment (akien-mga) [inline platform/android/java/app/config.gradle:10]: Do you know what's the state of Java 11 compatibility in the Android SDK / Studio?

Our documentation still tells people to install specifically JDK 8 for `jarsigner`, as earlier and later versions seemed to be problematic, and I remember running into issues with newer Java when using `sdkmanager`.

But many Linux distros are finally leaving OpenJDK 1.8 behind and defaulting to JDK 11 LTS, so we'll need to support this somehow if we can.

File: platform/android/java/app/config.gradle [hunk 2]
```diff
@@ -1,11 +1,11 @@
 ext.versions = [
-    androidGradlePlugin: '3.5.3',
+    androidGradlePlugin: '4.1.0',
     compileSdk         : 29,
     minSdk             : 18,
     targetSdk          : 29,
-    buildTools         : '29.0.3',
+    buildTools         : '30.0.1',
```
Comment (akien-mga) [inline platform/android/java/app/config.gradle:6]: I'm not super knowledgeable on how `buildTools` relate to the target SDK, but do we want to bump `compileSdk` and `targetSdk` to 30 eventually too?

And that's likely something we'll want to do in `3.2` too as usually Google makes this a requirement for new apps at some point. Judging by previous years, that would be from ~August 2021 (not announced yet): https://support.google.com/googleplay/android-developer/answer/113469#targetsdk

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

File: platform/android/java/app/config.gradle (hunk 2 above)
Comment (m4gr3d) [reply platform/android/java/app/config.gradle:6]: We need to resolve https://github.com/godotengine/godot/issues/38913 before we can bump the `targetSdk` and `compileSdk` as the [current workaround](https://developer.android.com/about/versions/11/privacy/storage#scoped-storage) we're using is ignored when targeting Android 11 (API level 30).

File: platform/android/java/app/config.gradle (hunk 1 above)
Comment (m4gr3d) [reply platform/android/java/app/config.gradle:10]: @akien-mga I'll look into it.

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.4.

**Comment by m4gr3d** [comment]
@akien-mga Is there a separate `3.2.4` branch? I don't see the changes made here in the `3.2` branch.

**Comment by akien-mga** [comment]
@m4gr3d I stage them locally and push them all as a batch after testing: #43456.

**Comment by m4gr3d** [comment]
@akien-mga Sounds good. I thought you had already done that step hence my confusion :)

**Pull request merged**